students/ChelseaSmith/lesson10/mailroom.py

```diff
@@ -0,0 +1,161 @@
+# functional mailroom program
+# Donor class holds individual donor information
+# Manager class accesses a text file storing all donors and manages that info
+# additional functions
+
+from functools import reduce
+
+class Donor:
+
+    def __init__(self, name, gifts):
+        self._name = name
+        self._donations = gifts
+
+    def append(self, addon):
+        self._donations.append(addon)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def donations(self):
+        return self._donations
+
+    def total_given(self):
+        total = sum(self._donations)
+        return total
+
+    def number_of_gifts(self):
+        number = len(self._donations)
+        return number
+
+    def average_gift(self):
+        ave = sum(self._donations)/len(self._donations)
+        return ave
+
+    def write_letter(self):
+        letter = "Dear {},\nThank you for your generous donation of ${:.2f}.\nSincerely,\n Generic Charities".format(self._name, self._donations[-1])
+        return letter
+
+    def multiply_gift(self, multiple):
+        newgift = [x * multiple for x in self._donations]
+        newdonor = Donor(self._name, newgift)
+        return newdonor
+
+
+
+class Manager:
+
+    def __init__(self, dictin): # changed this so I'm not pulling donor info from a file
+        self.record = list()
+        for k in dictin:
+            self.record.append(Donor(k, dictin[k]))
+
+    def __iter__(self):
+        return iter(self.record)
+
+    def append(self, addname, adddonation):
+        self.record.append(Donor(addname, adddonation))
+
+    def report(self):
+        print(f"Donor name" + " " * 7 + " | Total Given | Num Gifts | Average Gift\n")
+        for item in self.record:
+            print(f"{item.name:<18}${item.total_given():12.2f}{item.number_of_gifts():10}   ${item.average_gift():13.2f}\n")
+
+    def check_name(self, name):
+        new = False
+        for item in self.record:
+            if item.name == name:
+                new = True
+        return new
```

Comment:
note that you don't need the temp var `new`, you can control this with a return statement to simplify:
```
for item in self.record:
    if item.name == name:
        return True
else:
    return False